actions/delete_node_handler.go

```diff
@@ -52,12 +61,48 @@ func (h *deleteNodeHandler) Handle(ctx context.Context, action *castai.ClusterAc
 	log.Info("deleting kubernetes node")
 
 	b := backoff.WithContext(backoff.WithMaxRetries(backoff.NewConstantBackOff(h.cfg.deleteRetryWait), h.cfg.deleteRetries), ctx)
-	return backoff.Retry(func() error {
+	err := backoff.Retry(func() error {
 		err := h.clientset.CoreV1().Nodes().Delete(ctx, req.NodeName, metav1.DeleteOptions{})
 		if apierrors.IsNotFound(err) {
 			log.Info("node not found, skipping delete")
 			return nil
 		}
 		return err
 	}, b)
+
+	if err != nil {
+		return err
+	}
+
+	pods, err := h.clientset.CoreV1().Pods(metav1.NamespaceAll).List(ctx, metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": req.NodeName}).String(),
+	})
+	if err != nil {
+		return fmt.Errorf("listing node pods %w", err)
+	}
+
+	log.Infof("node has %d pods - removing", len(pods.Items))
+
+	deleteOptions := metav1.NewDeleteOptions(0)
+	deletePod := func(ctx context.Context, pod v1.Pod) error {
+		return h.deletePod(ctx, *deleteOptions, pod)
+	}
+
+	if err := h.sendPodsRequests(ctx, pods.Items, deletePod); err != nil {
+		return fmt.Errorf("sending delete pods requests: %w", err)
+	}
+
+	podsWait := backoff.WithContext(backoff.WithMaxRetries(backoff.NewConstantBackOff(h.cfg.podsTerminationWait), h.cfg.deleteRetries), ctx)
```

Comment:
25 seconds is enough for this operation?
Would be great to have these options configurable from env. We would have more flexibility if we would need to adjust the timeouts.

Reply:
Even 5s is enough for this operation, the node on which those pods are running is gone already.

We don't have env configuration options for any of  actions.  All actions have static config. I can introduce config from env but IMO this would be out of this PR.